notebooks/06a Modelling - classification.ipynb

```diff
@@ -0,0 +1,835 @@
+    "model_name = \"randomforest\"\n",
+    "\n",
+    "# define gridsearch parameters\n",
+    "gsc = GridSearchCV(\n",
+    "    estimator=RandomForestClassifier(class_weight=weights.to_dict()),\n",
+    "    param_grid={\"n_estimators\": [10, 100, 500], \"max_depth\": [5, 10, None]},\n",
+    "    cv=5,\n",
+    "    scoring=\"balanced_accuracy\",\n",
+    "    verbose=1,\n",
+    "    n_jobs=-1,\n",
+    "    refit=True,\n",
+    ")\n",
+    "\n",
+    "# takes ~8 mins to run on a STANDARD_DS3_V2\n",
+    "models[model_name] = train_model(gsc, X_train, y_train)\n",
+    "models[model_name]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Catboost\n",
+    "\n",
+    "Boosted tree optimised for categorical features"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1652706888015
+    }
+   },
+   "outputs": [],
+   "source": [
+    "model_name = \"catboost\"\n",
+    "\n",
+    "# extract categorical features\n",
+    "num_features = [\n",
+    "    \"AGE_ON_ADMISSION\",\n",
+    "    \"EL CountLast12m\",\n",
+    "    \"EMCountLast12m\",\n",
+    "    \"OP First CountLast12m\",\n",
+    "    \"OP FU CountLast12m\",\n",
+    "]\n",
+    "cat_features = list(set(X_train.columns) - set(num_features))\n",
+    "\n",
+    "# define gridsearch parameters\n",
+    "gsc = GridSearchCV(\n",
+    "    estimator=CatBoostClassifier(\n",
+    "        verbose=False, class_weights=weights.to_dict(), cat_features=cat_features\n",
+    "    ),\n",
+    "    param_grid={\n",
+    "        \"max_depth\": [5, 10, None],\n",
+    "        \"learning_rate\": [0.01, 0.1, 1],\n",
+    "        \"iterations\": [10, 100, 500],\n",
+    "    },\n",
+    "    cv=5,\n",
+    "    scoring=\"balanced_accuracy\",\n",
+    "    verbose=1,\n",
+    "    n_jobs=-1,\n",
+    "    refit=True,\n",
+    ")\n",
+    "\n",
+    "# takes ~25 mins to run on a STANDARD_DS3_V2\n",
+    "models[model_name] = train_model(gsc, X_train, y_train)\n",
+    "models[model_name]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### XGBoost"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1652706970050
+    },
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "model_name = \"xgboost\"\n",
+    "\n",
+    "# define gridsearch parameters\n",
+    "gsc = GridSearchCV(\n",
+    "    estimator=XGBClassifier(weight=weights.to_dict(), random_state=42),\n",
+    "    param_grid={\n",
+    "        \"n_estimators\": [1, 5],\n",
+    "        \"learning_rate\": [0.01, 0.1, 1],\n",
+    "        \"max_depth\": [5, 10, None],\n",
+    "    },\n",
+    "    cv=5,\n",
+    "    scoring=\"balanced_accuracy\",\n",
```

Comment:
Does using balanced accuracy increase the chance of False Positives as it focuses on Recall?